examples/complete-hosted-zone-in-diff-aws/main.tf

```diff
@@ -0,0 +1,64 @@
+provider "aws" {
+  region = "us-east-1"
+}
+
+# this is where hosted zone present/need to be created
+provider "aws" {
```

Comment:
I think further changes might be needed to illustrate a multi-account setup.

Neither this `provider` block nor the default provider block above (`provider "aws" {`) specify anything about authentication (for example, they have no [access_key](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#access_key), [secret_key](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#secret_key), or [profile](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#profile) arguments), so they will use `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` or `AWS_SHARED_CREDENTIALS_FILE` and `AWS_PROFILE` from the [environment](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#environment-variables). These will be the same throughout the execution of each Terraform command.

I think the best illustration of using the module cross-account would use the [assume_role](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#assume-role) argument. Assuming that in your example `terraform init`, `terraform plan`, and `terraform apply` would be run by someone with dev account credentials, the dev account user running `terraform apply` would be assuming a role in the prod account that had privileges to create the validation records. For this to really be an end-to-end example, as I believe module usage examples should be, the example configuration files would need to include the IAM role to be assumed.

Reply:
Yes, we can set the credentials in respective `aws` provider section using all the methods mentioned in terraform [documentation](https://registry.terraform.io/providers/hashicorp/aws/latest/docs). IMO, this would be the choice of the client of the module.

I have updated the example with use of `profile` with some documentation in `README` file of the example. Let me know if you have any other comments.